README.md

```diff
@@ -505,23 +532,22 @@ A good litmus test for invocation validity by a discharging agent is to check if
 
 ## 5.4 Rights Amplification
 
-Some capabilities are more than the sum of their parts. The canonical example is a can of soup and a can opener. You need both to access the soup inside the can, but the can opener may come from a completely separate source than the can of soup. Such semantics MAY be implemented in UCAN capabilities. This means that validating particular capabilities MAY require more than one direct witness. The relevant witnesses MAY be of a different resource and action from the amplified capability. The delegated capability MUST have this behaviour in its semantics, even if the witnesses do not.
+Some capabilities are more than the sum of their parts. The canonical example is a can of soup and a can opener. You need both to access the soup inside the can, but the can opener may come from a completely separate source than the can of soup. Such semantics MAY be implemented in UCAN capabilities. This means that validating particular capabilities MAY require more than one direct proof. The relevant proofs MAY be of a different resource and action from the amplified capability. The delegated capability MUST have this behaviour in its semantics, even if the proofs do not.
 
 ## 5.6 Content Identifiers
 
-UCANs MAY be referenced by content ID (CID), per the [multiformats/cid](https://github.com/multiformats/cid) specification. The resolution of these addresses is left to the implementation and end-user, and MAY (non-exclusively) includes the following: local store, a distributed hash table (DHT), gossip network, or RESTful service.
-
-CIDs MAY be used to refer to any UCAN: a witness in a delegation chain or an entire UCAN. This has many benefits, some of which are outlined in the implementation recommendations of this document.
+<!-- FIXME: resolution left up to implementer --> 
+UCAN proofs MUST be referenced by content ID (CID), per the [multiformats/cid](https://github.com/multiformats/cid) specification. The resolution of these addresses is left to the implementation and end-user, and MAY (non-exclusively) includes the following: local store, a distributed hash table (DHT), gossip network, or RESTful service.
 
-Due to the potential for unresolvable CIDs, this SHOULD NOT be the preferred method of transmission. "Inline witnesses" SHOULD be used whenever possible, and complete UCANs SHOULD be expanded. When a CID is used, it is RECOMMENDED that it be substituted as close to the top UCAN as possible (i.e. the invocation), and as few witnesses are referenced by CID, to keep the number of required CID resolutions to a minimum. As UCANs are signed, all further delegations would require CID resolution and so SHOULD NOT be used when the intention is delegation rather than invocation. 
+CIDs MAY be used to refer to any UCAN: a proof in a delegation chain or an entire UCAN. This has many benefits, some of which are outlined in the implementation recommendations of this document.
 
 ## 5.7 Revocation
```

Comment:
That's an interesting idea! 🤔

Reply:
Yeah the more I think about this the more I like it 👍 Let's put that in the draft and see how it feels / get feedback

Reply:
Ah, I'm starting to remember this now! We don't need all of the fields for a revocation, like an audience, expiry, and so on. All that we need to know in that message is:

1. Revoker's DID
2. CID to revoke
3. Signature

The format as currently specified is tuned to the use case, and avoids a couple weird ambiguities like the revocation expiring before the UCAN being revoked.

Reply:
I see makes sense. It makes me wonder if maybe invocation could also have different representation from other UCANs, because I don't find `nbf, exp, fct` to be useful and as opposed to `att` I would much rather have either a single capability been invoked or alternatively a dictionary of named invocations (kind of like graphql).

Reply:
Now that I come to think of it invocation in comparison to revocation only needs  `audience` and it might be nice if revocation would become invocation of built-in capability.

At the same time I think making invocation distinct from issuing / delegation would make them a lot more clearer that they currently are.

Reply:
After thinking about it a bit more, I'm realizing that decoupling invocation from issuing would have a side effect of not been able to derive another UCAN from invocation, which I'm divided on whether that would be a good thing. 

On one hand audience would not be able to do anything other than execute what's in the invocation, which kind of makes sense.

On the other hand we have couple of cases where invocation passes through multiple actors before it reaches actual service that will execute action and neither issuer nor actors in the middle necessarily know final audience. I do not know if that is mistake on our end caused by invocation & derivation been the same thing or whether having to distinguish between two would be a serious limitation.

Reply:
I think it depends on the use case. Something like a `ucan-rpc` pretty much say "do this now", but I can also imagine cases where you want to schedule a job, or run an ongoing cron. Should those be covered by an invocation interface, too?

Additionally, you likely _want_ to grant a full token along with the request to execute, since the recipient service may need to coordinate with other services to complete the request (e.g. send an email or iOS notification, which requires a sub-invocation).

Reply:
Should we write about the autonomous computing use case somewhere? services that invoke other services without pre-provisioning credentials

Reply:
@bmann some current wording on `main` intro section below. Lemme know if we should expand in the FAQ.

> Ahead-of-time coordination is often a barrier to development in many projects. Flexibility to define specialized authorization semantics for resources and the ability to integrate with external systems trustlessly are essential as the number of autonomous, specialized, and coordinating applications increases.

Reply:
kk. I’ll look at this and also look at spec vs ucan.xyz where copy like this can live.